src/types.ts

```diff
@@ -272,55 +272,85 @@ export interface Provider {
   ): Promise<ResolutionDetails<T>>;
 }
 
-export enum StandardResolutionReasons {
+export const StandardResolutionReasons = {
```

Comment:
I ended up using an object here because it's "less breaking" this way.

These are strings so we can still set them without referencing this structure.